### ISSUE TITLE
[YOLO] Fail to compile model when pretrained=False

### ISSUE THREAD

**Issue by vkovinicTT** [tenstorrent/tt-forge-fe#1466]
### Description
When trying to run the `test_yolov5_320x320` test with `pretrained=False`, Forge fails to compile the model even though it works perfectly fine when using `pretrained=True`. 

### Reproduction steps
```
git checkout vkovinic/yolo_fail
pytest -svv forge/test/models/pytorch/vision/yolo/test_yolo_v5.py::test_yolov5_320x320
```

1. Run the YOLOv5 test with `pretrained=True` - compiles and runs successfully
2. Run the same test with `pretrained=False` - fails during compilation

### Investigation
I've identified multiple differences between the pretrained and non-pretrained models:

1. BatchNorm layers: When using `pretrained=True`, the model automatically fuses BatchNorm layers with convolutions. With `pretrained=False`, BatchNorm layers remain separate, causing issues with TVM compilation.

2. When attempting to manually fuse BatchNorm layers on the non-pretrained model, it passes TVM but a different error in Forge occurs
